### DISCUSSION TITLE
Pre-enabling Modbus question if someone may explain something please? perhaps i have an issue i need to tell the tech coming?

### DISCUSSION THREAD

**Discussion by Menz01** [WillCodeForCats/solaredge-modbus-multi#271] [General]
Hello,
    Just had my solaredge system installed a couple of weeks ago and just turned on 2 days ago. Despite me asking the day of the installation, the techs had no clue what the "modbus over TCP" was and it was not enabled. i did however get him to run an ethernet cable to my router. I called and finally have a tech coming out next week to enable the modbus over TCP setting.
   
Here is my question though. i have 2 inverters and they appear to be setup with the two inverters wired together and one of the inverters has the cable runnung to my router. when i look at my router i see one inverter and depending on when i look at it on my router (i dont know the interval), the ip address will either be a routable IP address of 192.168.0.xxx like my network or it may be a 169.254.xxx.xxx address. according to Solaredge those are the IP's of the inverters. one has the 192 and the other has the 169. so my question or perhaps questions are... is this right? why would i see one inverter in my router (unifi UDM-Pro) but see the IP address swapping from time to time? also... as of right now the Intergration itself throws an error of not seeing the inverter and i am not sure if that is ok either? like i said i dont think the Modbus is enabled but i did try the other "HACS Modbus integration" and it installed and never threw an error but also showed everything as "unknown"

is there anything i should ask, say, try when the tech arrives next week to make sure this is a one time only visit? the bouncing IP addresses really concerns me the most.

thanks in advance
Mike

**Comment by WillCodeForCats** [comment]
Show the configuration instructions to the installer if they don't know where the modbus/tcp setting is.

You should statically assign the IP address in the inverter instead of using DHCP. If it changes randomly with DHCP you will have to remove and re-add the integration every time the IP changes.

**Reply by WillCodeForCats** [reply]
Or you can still use DHCP as long as you set a fixed client address in the router so it always assigns the same DHCP IP to the same client.

**Reply by Menz01** [reply]
so i actually have the IP for the inverter that grabs the 192.168.0.xxx DHCP reserved on a fixed IP address but what i am trying to figure out why the second inverter with the 169.254.xxx.xxx address shows in my router when only the one inverter is actually plugged in to my router via ethernet. the second inverter is only connected to the first inverter and does not have a wire coming back to the router

**Reply by rtlinux** [reply]
You have the correct setup, typically the leader would have Ethernet or 5G connected for the site and the followers are connected via RS-485-2 low voltage data cable (for firmware upgrades and communications etc...) 169.254.x.x just means it can't reach the DHCP server. It is not an issue, I think.

**Comment by Menz01** [comment]
so the tech came and said he enabled the modbus feature but i have a feeling something is still not right. i know i have 2 inverters but when i try and add the integration and telling it i have 2 inverters the setup fails but if i tell the setup only one inverter it configures just fine. so not sure what to do? does the enable modbus have to be enabled on both inverters? i know he told me he enabled it on both. is that the issue? something else perhaps?

**Reply by WillCodeForCats** [reply]
Follow all config steps: https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki/Configuration

**Reply by Menz01** [reply]
thanks but the issue is i could void my service warranty with the company that installed my equipment if i poke around the setup. so can you at least answer my question... since i know he enabled the Modbos on both inverters and i think your documentation that you are referring me to says it only has to be done on one. is that going to cause an issue?

**Reply by WillCodeForCats** [reply]
no

**Reply by WillCodeForCats** [reply]
If you aren't seeing both inverters the setup is incorrect. If your installer isn't able to figure it out then you'll just have to live with only one inverter being usable unless it can be set up according to the instructions.

**Comment by Menz01** [comment]
ok... so i took a chance and went outside and connected to my first inverter and i can confirm the modbus is enabled and and the device ID is 1. However... even though i conneced to the wifi the same way as i connected to the first inverter, the webpage 172.16.0.1 would not open so i could not see anything on that one. the meter settings also looked right on the first inverter as well

**Reply by WillCodeForCats** [reply]
Usually the problem is the unit id on the second inverter is not set correctly.

**Reply by Menz01** [reply]
isn't odd though that i could connect to the inverters wifi but could not open a web page to look at it?
